cmd/proxy/main.go

```diff
@@ -409,12 +408,32 @@ func createVersionsRegistry(
 		return nil, err
 	}
 
-	shardCoord, err := sharding.NewMultiShardCoordinator(cfg.GeneralSettings.NumberOfShards, 0)
+	httpClient := &http.Client{}
+	httpClient.Timeout = time.Duration(cfg.GeneralSettings.RequestTimeoutSec) * time.Second
+	observersList := make([]string, 0, len(cfg.Observers))
+	for _, node := range cfg.Observers {
+		observersList = append(observersList, node.Address)
+	}
+	argsNumShardsProcessor := process.ArgNumShardsProcessor{
+		HttpClient:                    httpClient,
+		Observers:                     observersList,
+		TimeBetweenNodesRequestsInSec: cfg.GeneralSettings.TimeBetweenNodesRequestsInSec,
+		NumShardsTimeoutInSec:         cfg.GeneralSettings.NumShardsTimeoutInSec,
+		RequestTimeoutInSec:           cfg.GeneralSettings.RequestTimeoutSec,
+	}
+	numShardsProcessor, err := process.NewNumShardsProcessor(argsNumShardsProcessor)
+	if err != nil {
+		return nil, err
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	numShards, err := numShardsProcessor.GetNetworkNumShards(ctx)
+	cancel()
```

Comment:
maybe we can extract this into a separate func for getting the number of shards, since the component is created only for this usage

Reply:
pushed